plugin/cmake.vim

```diff
@@ -68,6 +68,15 @@ function! s:find_build_dir()
     return 1
   endif
 
+  if exists("g:cmake_build_dir_override")
```

Comment:
I will name this `g:cmake_create_build_dir` and would allow values `0` (for off, default) or `git-root`, which would create the build dir (`g:cmake_build_dir`) in the git root as proposed. I could imaging some `toplevel-cmakelists` value in the future for example, which would allow searching for the toplevel CMakeLists.txt file.

Would that work for you? :)